miner/payload_building.go

```diff
@@ -231,38 +313,61 @@ func (w *worker) buildPayload(args *BuildPayloadArgs) (*Payload, error) {
 		timer := time.NewTimer(0)
 		defer timer.Stop()
 
+		start := time.Now()
+		const blockTime = 2 * time.Second // OP default block time
 		// Setup the timer for terminating the process if SECONDS_PER_SLOT (12s in
 		// the Mainnet configuration) have passed since the point in time identified
 		// by the timestamp parameter.
-		endTimer := time.NewTimer(time.Second * 12)
+		endTimer := time.NewTimer(blockTime)
+		defer endTimer.Stop()
```

Comment:
This code-path shouldn't be dependent on a constant block-time, especially since it's also used for L1 testing in op-e2e and possibly elsewhere.

Instead, can we make `validateParams` return the block time? It already looks up the parent block header (which includes the timestamp), and has the payload arguments with timestamp, so it can compute the difference. And then maybe clip it to a minimum of `2s` (in case the block-time is unreasonably low, or instant).